src/libs/actions/PersonalDetails.js

```diff
@@ -186,6 +199,36 @@ function getFromReportParticipants(reports) {
         });
 }
 
+
+/**
+ * Sets the personal details object for the current user
+ *
+ * @param {Object} details
+ */
+function setPersonalDetails(details) {
+    API.PersonalDetails_Update({details: JSON.stringify(details)});
+    NameValuePair.set('timezone', details.timezone);
+
+    // Update the associated onyx keys
+    Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, details);
+    Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, formatPersonalDetails({[currentUserEmail]: details}));
+}
+
+/**
+ * Sets the user's avatar image
+ *
+ * @param {String} base64image
+ */
+function setAvatar(base64image) {
+    API.User_UploadAvatar({base64image}).then((response) => {
```

Comment:
are we preferring `then` instead of `done`, `fail`, `always`?

Reply:
Looks like there are significantly more usages of `.then` (87 results) vs `.done` (4 results), `.fail` (2 results) and `.always` (0 results)

Reply:
We are using the `fetch()` API. `.done()` is used for the jQuery Deferred object style of a promise which we do not use in `Expensify.cash`

Reply:
ah cool cool, I figured it was something like that. Thanks marc!